src/scripts/renderPages.js

```diff
@@ -0,0 +1,132 @@
+);
+
+const checkCounterForRightPagination = (currentElement, maxElement) => {
+    return Math.min(currentElement - 2, maxElement - 3);
+};
+
+const checkCounterForLeftPagination = currentElement => {
+    return Math.max(currentElement + 1, 4);
+};
+
+const checkCounterForMiddlePagination = currentElement => {
+    return currentElement - 2;
+};
+
+const checkPaginationEnd = paginationStart => {
+    return paginationStart + 3;
+};
+
+const appendFirstElements = () => {
+    const firstElements = /* HTML */ `
+        <div class="pagination-component__page">1</div>
+        <div
+            class="
+            pagination-component__page"
+        >
+            ...
+        </div>
+    `;
+    paginationComponentPages.insertAdjacentHTML('afterbegin', firstElements);
+};
+
+const appendLastElements = maxElement => {
+    const lastElements = /* HTML */ `
+        <div class="pagination-component__page">...</div>
+        <div
+            class="
+            pagination-component__page"
+        >
+            ${maxElement}
+        </div>
+    `;
+    paginationComponentPages.insertAdjacentHTML('beforeend', lastElements);
+};
+
+const createPaginationElements = (
+    paginationStart,
+    paginationEnd,
+    currentElement
+) => {
+    let paginationPages = ``;
+
+    for (let counter = paginationStart; counter <= paginationEnd; counter++) {
+        if (currentElement === counter) {
+            paginationPages += `<div
+            class="pagination-component__page pagination-component__page--active">${counter}</div>`;
+            continue;
+        }
+        paginationPages += `<div
+            class="pagination-component__page">${counter}</div>`;
+    }
+    paginationComponentPages.insertAdjacentHTML('beforeend', paginationPages);
+};
+
+const createLeftActiveSlides = (currentElement, maxElement) => {
+    const paginationEnd = checkCounterForLeftPagination(currentElement);
+    createPaginationElements(1, paginationEnd, currentElement);
+    appendLastElements(maxElement);
+};
+
+const createRightActiveSlides = (currentElement, maxElement) => {
+    const paginationStart = checkCounterForRightPagination(
+        currentElement,
+        maxElement
+    );
+    createPaginationElements(paginationStart, maxElement, currentElement);
+    appendFirstElements();
+};
+
+const createMiddleActiveSlides = (currentElement, maxElement) => {
+    const paginationStart = checkCounterForMiddlePagination(currentElement);
+    const paginationEnd = checkPaginationEnd(paginationStart);
+    createPaginationElements(paginationStart, paginationEnd, currentElement);
+    appendFirstElements();
+    appendLastElements(maxElement);
+};
+
+const isElementBiggerThanFour = currentElement => {
+    return currentElement > 4;
+};
+
+const isElementLessThanMaxMinusThree = (currentElement, maxElement) => {
+    return currentElement < maxElement - 2;
+};
+
+const isMaxElementLessOrEqualThanSix = maxElement => {
+    return maxElement <= 6;
+};
```

Comment:
nope... xd 
Co to jest to 4? zdefiniuj to jakoś i w kontekście tego nazwij funkcje xd Taka nazwa jest bez sensu + jak zmienisz implementacje funkcji to nagle nazwa jest z dupy.
Kolejne metody to samo 
szczególnie że w `isElementLessThanMaxMinusThree` widzę `maxElement - 2` 😂